src/variables.rs

```diff
@@ -0,0 +1,72 @@
+use yultsur::yul::{Identifier, IdentifierID};
+
+use crate::smt::{self, SMTExpr, SMTSort, SMTStatement, SMTVariable};
+use crate::ssa_tracker::SSATracker;
+
+pub struct Variable {
+    pub identifier: Identifier,
+    pub domain: Vec<SMTSort>,
+    pub codomain: SMTSort,
+    pub default_value: Option<SMTExpr>,
```

Comment:
This could be a bit of a hindrance later, but I would say let's keep it for now, we can always refactor further.